src/zoid/buttons/component.jsx

```diff
@@ -681,6 +682,62 @@ export const getButtonsComponent: () => ButtonsComponent = memoize(() => {
         },
       },
 
+      onMessageHover: {
+        type: "function",
+        required: false,
+        value: () => {
+          return () => {
+            // lazy loads the modal, to be memoized and executed onMessageClick
+            getModal();
+          };
+        },
+      },
+
+      onMessageClick: {
+        type: "function",
+        required: false,
+        value: ({ props }) => {
+          return async ({ offerType, messageType }) => {
+            const {
+              message,
+              clientID,
+              merchantID,
+              currency,
+              locale: { lang, country },
+              buttonSessionID,
+            } = props;
+            const amount = message?.amount || undefined;
+
+            const modal = await getModal();
+            const modalInstance = modal({
+              account: `client-id:${clientID}`,
+            });
+            modalInstance.show({
+              amount,
+              offer: offerType?.join(",") || undefined,
+              account: `client-id:${clientID}`,
+              merchantId: merchantID?.join(",") || undefined,
+              language: `${lang}-${country}`,
+              buyerCountry: getBuyerCountry(),
+              currency,
+            });
+
+            getLogger()
+              .info("button_message_clicked")
+              .track({
+                [FPTI_KEY.EVENT_NAME]: "message_click",
+                // [FPTI_KEY.BUTTON_MESSAGE_OFFER_TYPE]: offerType,
+                // [FPTI_KEY.BUTTON_MESSAGE_TYPE]: messageType,
+                // [FPTI_KEY.BUTTON_MESSAGE_POSITION]: message.position,
+                // [FPTI_KEY.BUTTON_MESSAGE_ALIGN]: message.align,
+                // [FPTI_KEY.BUTTON_MESSAGE_COLOR]: message.color,
+                // [FPTI_KEY.AMOUNT]: amount,
```

Comment:
@paypal-rosman I entered the FPTI_KEYs for you, and they are commented out until you get to add them to the sdk-constants

Reply:
@danzhaaspaypal I we also want `MESSAGE_TYPE` here as well? @Seavenly thoughts ?

Reply:
@paypal-rosman I think I've got you covered on line 730 there, unless you're requesting a name change?

Reply:
my eyes are still not working lol thank you